commit b8af7a0eef2ae0146f190c6c8df542e542d6f8e0
Author: Stefan Wahren <stefan.wahren@i2se.com>
Date:   Mon Oct 22 15:16:51 2018 +0200

    staging: vchiq_arm: Fix camera device registration
    
    Since the camera driver isn't probed via DT, we need to properly setup DMA.
    
    Fixes: 37b7b3087a2f ("staging/vc04_services: Register a platform device for the camera driver.")
    Signed-off-by: Stefan Wahren <stefan.wahren@i2se.com>

diff --git a/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c b/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
index 7ab3c29c03a5..e62222549523 100644
--- a/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
+++ b/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
@@ -49,6 +49,7 @@
 #include <linux/of.h>
 #include <linux/platform_device.h>
 #include <linux/compat.h>
+#include <linux/dma-mapping.h>
 #include <soc/bcm2835/raspberrypi-firmware.h>
 
 #include "vchiq_core.h"
@@ -3578,6 +3579,21 @@ void vchiq_platform_conn_state_changed(VCHIQ_STATE_T *state,
 	}
 }
 
+static struct platform_device *
+vchiq_register_child(struct platform_device *pdev, const char *name)
+{
+	struct platform_device_info pdevinfo;
+
+	memset(&pdevinfo, 0, sizeof(pdevinfo));
+
+	pdevinfo.parent = &pdev->dev;
+	pdevinfo.name = name;
+	pdevinfo.id = PLATFORM_DEVID_NONE;
+	pdevinfo.dma_mask = DMA_BIT_MASK(32);
+
+	return platform_device_register_full(&pdevinfo);
+}
+
 static int vchiq_probe(struct platform_device *pdev)
 {
 	struct device_node *fw_node;
@@ -3637,9 +3653,7 @@ static int vchiq_probe(struct platform_device *pdev)
 		VCHIQ_VERSION, VCHIQ_VERSION_MIN,
 		MAJOR(vchiq_devid), MINOR(vchiq_devid));
 
-	bcm2835_camera = platform_device_register_data(&pdev->dev,
-						       "bcm2835-camera", -1,
-						       NULL, 0);
+	bcm2835_camera = vchiq_register_child(pdev, "bcm2835-camera");
 
 	return 0;
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1231_b8af7a0eef2ae0146f190c6c8df542e542d6f8e0.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1231_b8af7a0eef2ae0146f190c6c8df542e542d6f8e0.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
